src/pretix/plugins/socialauth/backends.py

```diff
@@ -0,0 +1,51 @@
+from allauth.socialaccount.adapter import get_adapter
+
+from pretix.base.auth import BaseAuthBackend
+from pretix.helpers.urls import build_absolute_uri
+
+adapter = get_adapter()
+
+
+class MediaWikiBackend(BaseAuthBackend):
+    identifier = 'mediawiki'
+
+    @property
+    def verbose_name(self):
+        return "Login with MediaWiki"
+
+    def authentication_url(self, request):
+        return (
+            adapter.get_provider(request, 'mediawiki').get_login_url(request)
+            + "?next="
```

Comment:
Don't build URL by directly concatenating path and queries like this.
Use `urlencode` from [`urllib.parse`](https://docs.python.org/3/library/urllib.parse.html)